app/src/main/java/com/pvp/app/service/HealthConnectServiceImpl.kt

```diff
@@ -44,4 +47,22 @@ class HealthConnectServiceImpl @Inject constructor(
             0L
         }
     }
+
+    override suspend fun aggregateDistance(
+        start: Instant,
+        end: Instant
+    ): Double {
+        return try {
+            val response = client.aggregate(
+                AggregateRequest(
+                    metrics = setOf(DistanceRecord.DISTANCE_TOTAL),
+                    timeRangeFilter = TimeRangeFilter.between(start, end)
+                )
+            )
+            response[DistanceRecord.DISTANCE_TOTAL]?.inMeters ?: 0.0
+        } catch (e: Exception) {
+            Log.e("DISTANCE", "EXCEPTION ${e.toString()}")
```

Comment:
`{e}` would automatically be converted to string when used in a string interpolation/concatation, if I recall correctly. Or if a message is all you need, use `${e.message}` then.